xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/DeleteAction.java

```diff
@@ -147,8 +151,9 @@ protected boolean delete(XWikiContext context) throws XWikiException
             sendRedirect(response, Utils.getRedirect("view", context));
             return true;
         } else {
-            // Delete to recycle bin.
-            return deleteToRecycleBin(context);
+            // The document skips the recycle bin only if the user has explicitly made the choice to skip it.
```

Comment:
And only if the user is allowed to skip it no? That will be checked downstream AFAIU.

So maybe the text should be rephrased to mention that it's a wish to skip the recycle bin or not but that it'll be enforced downstream depending if the user is allowed or not?

Reply:
And documentation updated https://github.com/xwiki/xwiki-platform/pull/1401/files#diff-70855079f42aabba98e87755fa9e71ecR155